Src/IronPython/Runtime/Operations/LongOps.cs

```diff
@@ -14,106 +14,218 @@
 using Microsoft.Scripting.Runtime;
 using Microsoft.Scripting.Utils;
 
+using IronPython.Runtime.Exceptions;
 using IronPython.Runtime.Types;
 
 namespace IronPython.Runtime.Operations {
 
     public static partial class BigIntegerOps {
-        [StaticExtensionMethod]
-        public static object __new__(CodeContext context, PythonType cls, string s, int radix) {
-            var value = LiteralParser.ParseBigIntegerSign(s, radix, Int32Ops.FindStart(s, radix));
 
-            if (cls == TypeCache.BigInteger) {
-                return value;
-            } else {
-                return cls.CreateInstance(context, value);
+        #region Constructors
+
+        private static object FastNew(CodeContext/*!*/ context, object o, int @base = 10) {
```

Comment:
Yes, I was planning to handle it in the follow-up `LongOps`/`IntOps` unification. It is kind of a separate issue since there are several ways of doing it. The idea I had was to make `BigIntegerOps` the main version and call it from `Int32Ops` as appropriate. This also applies to other methods in `Int32Ops`.